`.secretlintignore` does not work in Windows

<!--
Please update secretlint and try it before bug report.
-->

**What version of secretlint are you using?**
<!-- 
# Past result of following command: 
$ npx secretlint --version
-->
7.0.3

**What did you do? Please include the actual steps causing the issue.**
1. Write a secret in `test.txt`
2. Add it to `.secretlintignore`
```
test.txt
```
3. Run `pnpm secretlint "**/*" --maskSecrets`
<!-- 
Reproduce repository is helpful for debugging! 
-->

**What did you expect to happen?**
No errors were thrown.

**What actually happened? Please include the actual, raw output from secretlint.**
<!--
You can get debug log by running secretlint with `--debug` option.
$ npx secretlint --debug "target/**/*"
# Please paste the debug log to the issue or use http://gist.github.com/
-->
`test.txt` has not been ignored.

```
  secretlint input: [ '**/*' ] +0ms
  secretlint flags: {
  secretlint   maskSecrets: true,
  secretlint   init: false,
  secretlint   format: 'stylish',
  secretlint   secretlintignore: '.secretlintignore',
  secretlint   color: true,
  secretlint   terminalLink: true,
  secretlint   profile: false,
  secretlint   cwd: 'C:\\Users\\taku\\Documents\\GitHub\\reponame',
  secretlint   debug: false
  secretlint } +5ms
  secretlint searchFiles ignore baseDir: , normalizeIgnoreFilePath: C:\Users\taku\Documents\GitHub\reponame\.secretlintignore +0ms
  secretlint ignored: [ '/Users/taku/Documents/GitHub/reponame/test.txt' ] +2ms
  secretlint search patterns: [ '**/*' ] +1ms
  secretlint search ignore patterns: [ '**/.git/**', '**/node_modules/**', '**/.secretlintrc/**', '**/.secretlintrc.{json,yaml,yml,js}/**', '**/.secretlintignore*/**', '/Users/taku/Documents/GitHub/reponame/test.txt' ] +0ms

C:/Users/taku/Documents/GitHub/reponame/test.txt
  1:0  error  [GITHUB_TOKEN] found GitHub Token(*****************************): ****************************************  @secretlint/secretlint-rule-preset-recommend > @secretlint/secretlint-rule-github

✖ 1 problem (1 error, 0 warnings)
```

I will create a PR to fix this issue soon.

--- a/packages/secretlint/src/search.ts
+++ b/packages/secretlint/src/search.ts
@@ -16,11 +16,10 @@ export type SearchFilesOptions = {
 };
 
 const mapGitIgnorePatternTo = (base: string) => (ignore: string) => {
-    if (ignore.startsWith("!")) {
-        return "!" + path.posix.resolve(path.posix.join(base, ignore.slice(1)));
-    }
-
-    return path.posix.resolve(path.posix.join(base, ignore));
+    const mapped = ignore.startsWith("!")
+        ? "!" + path.resolve(path.join(base, ignore.slice(1)))
+        : path.resolve(path.join(base, ignore));
+    return mapped.replace(/\\/g, "/");
 };
 /**
  * globby wrapper that support ignore options

